### ISSUE TITLE
Source Gitlab: support oauth

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#6264]
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application. 

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. I suggest that this be a oneof nested inside a top-level field called "authentication": 

`
{
  authentication: {
     type: object
     oneOf: [ 
       // api key,
       // oauth
     ] 
  }
}
`

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change. 

## Acceptance Criteria

1. The connector supports oauth webflow authentication with client_id/client_secret 
2. Oauth properties are annotated properly. See [this PR](https://github.com/airbytehq/airbyte/pull/6022) for an example

**Comment by YowanR** [comment]
@igrankova @ycherniaiev Can you confirm that this work was blocked on Single Refresh Token? Thanks

**Comment by igrankova** [comment]
@YowanR 
I don't see it's blocked. We have the public app and creds, everything looks fine from our side.
[Public App](https://gitlab.com/oauth/applications/202383)
Client and Secret Lastpass name: GitLab Public Oauth Developer Application
Account LastPass name: gitlab.com
[OAuth docs](https://docs.gitlab.com/ee/api/oauth2.html#authorization-code-flow)
Let me know if I didn't get the question.
cc @ycherniaiev

**Comment by davydov-d** [comment]
Closing it as a duplicate of https://github.com/airbytehq/airbyte/issues/7480
